### PR TITLE
ci-e2e: backport changes in conformance-e2e into v1.13 tests 

### PR DESCRIPTION
**Pull request by brb**
To unblock https://github.com/cilium/cilium/pull/25346 let's backport the whole conformance-e2e.yaml to conformance-e2e-v1.13.yaml.

### CODE REVIEW

**Comment by thorn3r** [comment]
@brb right forgot about that, ty!

**Review by thorn3r — APPROVED** [review]

File: .github/workflows/conformance-e2e-v1.13.yaml [hunk 1]
```diff
@@ -20,7 +20,7 @@ on:
   #   will disappear from the PR checks: please provide a direct link to the
   #   successful workflow run (can be found from Actions tab) in a comment.
   #
-  # pull_request: {}
+  pull_request: {}
```
Comment (tommyp1ckles) [inline .github/workflows/conformance-e2e-v1.13.yaml]: should this remain commented?

Comment (thorn3r) [reply .github/workflows/conformance-e2e-v1.13.yaml]: nope this was just to get the tests to run here

**Comment by thorn3r** [comment]
/ci-e2e-1.13

**Comment by thorn3r** [comment]
@brb feel free to drop this commit if it's not what you had in mind. I can also move this to a new PR instead.

**Comment by brb** [comment]
@thorn3r Cool, that was my intention for a separate PR, as I didn't want to introduce that many changes to the already broken flow. But considering that all tests have passed (https://github.com/cilium/cilium/actions/runs/4955234361/jobs/8864455590?pr=25386), let's proceed with your changes :tada: I applied a few small changes to your commit and cleaned up the rest.

**Comment by brb** [comment]
Latest run (all green): https://github.com/cilium/cilium/actions/runs/4955981127/jobs/8865913448

**Review by YutaroHayakawa — APPROVED** [review]

**Pull request merged**